src/main/js/script/EditForm.tsx

```diff
@@ -21,230 +21,130 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import React from "react";
-import { WithTranslation, withTranslation } from "react-i18next";
+import React, { FC, FormEvent, useState } from "react";
+import { useTranslation } from "react-i18next";
 import {
-  InputField,
-  LabelWithHelpIcon,
-  ButtonGroup,
   Button,
-  SubmitButton,
-  Textarea,
+  ButtonGroup,
   ErrorNotification,
+  InputField,
+  LabelWithHelpIcon,
+  Level,
   Notification,
-  Level
+  SubmitButton,
+  Textarea
 } from "@scm-manager/ui-components";
-import { modify, run } from "../api";
+import { useRunScript, useUpdateScript } from "../api";
 import { Script, ScriptExecutionResult, ScriptLinks } from "../types";
 import ContentEditor from "../components/ContentEditor";
 import Output from "../components/Output";
 import DeleteScript from "./DeleteScript";
 
-type Props = WithTranslation & {
+type Props = {
   script: Script;
-  onDelete: () => void;
   links: ScriptLinks;
 };
 
-type State = Script & {
-  loading: boolean;
-  error?: Error;
-  result?: ScriptExecutionResult;
-  saveSuccess: boolean;
-};
-
-class EditForm extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      ...props.script,
-      loading: false,
-      saveSuccess: false
-    };
-  }
-
-  onRun = (event: Event) => {
-    this.onAction(event);
-
-    const { script, links } = this.props;
-    const { content } = this.state;
-
-    if (!links.execute) {
-      return;
-    }
-
-    run(links.execute, script.type, content ? content : "")
-      .then(result =>
-        this.setState({
-          loading: false,
-          error: undefined,
-          result
-        })
-      )
-      .catch(error =>
-        this.setState({
-          loading: false,
-          error,
-          result: undefined
-        })
-      );
+const EditForm: FC<Props> = ({ script, links }) => {
+  const [t] = useTranslation("plugins");
+  const [scriptState, setScriptState] = useState<Script>(script);
+  const [result, setResult] = useState<ScriptExecutionResult | undefined>();
+  const [saveSuccess, setSaveSuccess] = useState(false);
+  const { error: updateError, isLoading: updateLoading, update } = useUpdateScript(script);
+  const { error: runError, isLoading: runLoading, run } = useRunScript(links.execute!, r => setResult(r));
+
+  const onRun = async (event: FormEvent<HTMLFormElement>) => {
```

Comment:
Is the typing correct? It does not match the button action right any why it is async?